### PR TITLE
[Bug Fix] PositiveDetectionsVSDetectionEngines error handling

### PR DESCRIPTION
**Pull request by DeanArbel**
## Status
Ready

## Description
Fixed an issue that made the script display error messages when it didn't get required fields.

## Does it break backward compatibility?
   - No

### No custom fields / No Detection Engines + No Postive Detections
image

#### Positivedetections missing
image

#### Detecion Engines < Positive Detections
image

## Must have
- [x] Tests
- [ ] Documentation (with link to it)
- [ ] Code Review

## Technical writer review
Mention and link to the files that require a technical writer review.
- [ ] [CHANGELOG](https://github.com/demisto/content/edit/psd-fix/Scripts/PositiveDetectionsVSDetectionEngines/CHANGELOG.md?pr=%2Fdemisto%2Fcontent%2Fpull%2F5455)

### CODE REVIEW

File: Scripts/PositiveDetectionsVSDetectionEngines/PositiveDetectionsVSDetectionEngines.py [hunk 1]
```diff
@@ -4,50 +4,74 @@
...
+
+
 def main():
-    indicator_data = demisto.args().get('indicator')
-    demisto.results(extract_engines_data_from_indicator(indicator_data))
+    try:
+        indicator_data = demisto.args().get("indicator")
+        demisto.results(extract_engines_data_from_indicator(indicator_data))
+    except Exception as e:
+        demisto.debug(f"PostiveDetectionsVSDetectiongEngines failed with [{e}]")
```
Comment (anara123) [inline Scripts/PositiveDetectionsVSDetectionEngines/PositiveDetectionsVSDetectionEngines.py]:
```suggestion
        demisto.error(f"PostiveDetectionsVSDetectiongEngines failed with [{e}]")
```

File: Scripts/PositiveDetectionsVSDetectionEngines/PositiveDetectionsVSDetectionEngines.py [hunk 2]
```diff
@@ -4,50 +4,74 @@
...
+    if not indicator_data:
+        exit_with_message("No indicator found")
+    cstm_fields = indicator_data.get("CustomFields")
+    if not cstm_fields:
+        # No content, so will display 0/0
+        return create_pie(0, 0)
+    elif "detectionengines" not in cstm_fields:
+        exit_with_message('Please provide  Custom Field "Detection Engines"')
+    elif "positivedetections" not in cstm_fields:
+        exit_with_message('Please provide  Custom Field "Positive Detections"')
```
Comment (anara123) [inline Scripts/PositiveDetectionsVSDetectionEngines/PositiveDetectionsVSDetectionEngines.py]: where the user will see this messages? `'Please provide  Custom Field "Positive Detections"'`

File: Scripts/PositiveDetectionsVSDetectionEngines/PositiveDetectionsVSDetectionEngines.py [hunk 3]
```diff
@@ -4,50 +4,74 @@
...
+        },
     }
-
     return data
 
 
+def exit_with_message(msg="Failed to load"):
+    msg = f"Could not load widget:\n{msg}"
+    demisto.results(msg)
+    sys.exit(0)
```
Comment (anara123) [inline Scripts/PositiveDetectionsVSDetectionEngines/PositiveDetectionsVSDetectionEngines.py]: it is a bad practice to sys.exit(0)

Comment (DeanArbel) [reply Scripts/PositiveDetectionsVSDetectionEngines/PositiveDetectionsVSDetectionEngines.py]: removed

File: Scripts/PositiveDetectionsVSDetectionEngines/PositiveDetectionsVSDetectionEngines.py (hunk 1 above)
Comment (DeanArbel) [reply Scripts/PositiveDetectionsVSDetectionEngines/PositiveDetectionsVSDetectionEngines.py]: changed

File: Scripts/PositiveDetectionsVSDetectionEngines/PositiveDetectionsVSDetectionEngines.py [hunk 4]
```diff
@@ -4,50 +4,74 @@
...
-                        detection_engines - positive_detections
-                    ],
+                    "data": [unknown_detections],
                     "groups": None,
                     "name": "Unknown",
                     "label": "Unknown",
-                    "color": "rgb(255, 144, 0)"
-                }
+                    "color": "rgb(255, 144, 0)",
+                },
```
Comment (anara123) [inline Scripts/PositiveDetectionsVSDetectionEngines/PositiveDetectionsVSDetectionEngines.py]:
```suggestion
                }
```

File: Scripts/PositiveDetectionsVSDetectionEngines/PositiveDetectionsVSDetectionEngines.py [hunk 5]
```diff
@@ -4,50 +4,74 @@
...
-                    "data": [
-                        detection_engines - positive_detections
-                    ],
+                    "data": [unknown_detections],
                     "groups": None,
                     "name": "Unknown",
                     "label": "Unknown",
-                    "color": "rgb(255, 144, 0)"
-                }
+                    "color": "rgb(255, 144, 0)",
```
Comment (anara123) [inline Scripts/PositiveDetectionsVSDetectionEngines/PositiveDetectionsVSDetectionEngines.py]:
```suggestion
                    "color": "rgb(255, 144, 0)"
```

File: Scripts/PositiveDetectionsVSDetectionEngines/PositiveDetectionsVSDetectionEngines.py [hunk 6]
```diff
@@ -4,50 +4,74 @@
...
                 {
-                    "data": [
-                        positive_detections
-                    ],
+                    "data": [positive_detections],
                     "groups": None,
                     "name": "Positive Detections",
                     "label": "Positive Detections",
-                    "color": "rgb(255, 23, 68)"
+                    "color": "rgb(255, 23, 68)",
```
Comment (anara123) [inline Scripts/PositiveDetectionsVSDetectionEngines/PositiveDetectionsVSDetectionEngines.py]:
```suggestion
                    "color": "rgb(255, 23, 68)"
```

File: Scripts/PositiveDetectionsVSDetectionEngines/PositiveDetectionsVSDetectionEngines.py [hunk 7]
```diff
@@ -4,50 +4,74 @@
...
-                    "color": "rgb(255, 144, 0)"
-                }
+                    "color": "rgb(255, 144, 0)",
+                },
             ],
-            "params": {
-                "layout": "vertical"
-            }
-        }
+            "params": {"layout": "vertical"},
```
Comment (anara123) [inline Scripts/PositiveDetectionsVSDetectionEngines/PositiveDetectionsVSDetectionEngines.py]:
```suggestion
            "params": {"layout": "vertical"}
```

**Comment by anara123** [comment]
@amshamah419 please review the one liner release notes so we could merge

**Pull request merged**